## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#37effcae-a46d-4a76-8b9a-ad9d34cdd963) |
| mint-dist-erasure.sh | :x: [more...](#2f48a660-9f14-4767-93a8-ac244be8ff74) |
| mint-zoned.sh | :x: [more...](#4b2392c7-44ab-485c-9205-a0e2eceba68d) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#0298852f-f34d-45d7-992d-e028d7287cea) |
<a name="0298852f-f34d-45d7-992d-e028d7287cea"></a>
### 12659-0b4ef68/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s2:32551
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/16) Running aws-sdk-go tests ... FAILED in 6 minutes and 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-mmpupgt9bjagl3",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 366258,
  "error": "SlowDown: Resource requested is unreadable, please reduce your request rate\n\tstatus code: 503, request id: 1690256C08CD7202, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go DeleteBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/16) Running aws-sdk-java tests ... FAILED in 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-mmpupgt9bjagl3",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 366258,
  "error": "SlowDown: Resource requested is unreadable, please reduce your request rate\n\tstatus code: 503, request id: 1690256C08CD7202, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go DeleteBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/16) Running aws-sdk-php tests ... FAILED in 4 seconds
{
  "name": "aws-sdk-php",
  "function": "getBucketPolicy ( array $params = [] )",
  "args": {
    "Bucket": "aws-sdk-php-60e85be7462fd"
  },
  "duration": "268",
  "status": "FAIL",
  "error": "Error executing \"DeleteBucket\" on \"http://minio-k8s2:32551/aws-sdk-php-60e85be7462fd\"; AWS HTTP error: Server error: `DELETE http://minio-k8s2:32551/aws-sdk-php-60e85be7462fd` resulted in a `503 Service Unavailable` response:\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>SlowDown</Code><Message>Resource requested is unreadable, please red (truncated...)\n SlowDown (server): Resource requested is unreadable, please reduce your request rate - <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>SlowDown</Code><Message>Resource requested is unreadable, please reduce your request rate</Message><BucketName>aws-sdk-php-60e85be7462fd</BucketName><Resource>/aws-sdk-php-60e85be7462fd</Resource><RequestId>1690256DE1119EB8</RequestId><HostId>f8249e62-3420-4382-b112-4f8a1dd0dbb2</HostId></Error>"
}
(1/16) Running aws-sdk-ruby tests ... FAILED in 3 seconds
{
  "name": "aws-sdk-ruby",
  "function": "removeBucket(bucket_name)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-528affdd152a"
  },
  "duration": 2271.42,
  "error": "Failed to clean-up bucket 'aws-sdk-ruby-bucket-528affdd152a', Resource requested is unreadable, please reduce your request rate",
  "status": "FAIL"
}
(1/16) Running awscli tests ... FAILED in 19 seconds
{
  "name": "awscli",
  "duration": 17204,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "remove_bucket failed: s3://awscli-mint-test-bucket-21044 An error occurred (SlowDown) when calling the DeleteBucket operation (reached max retries: 4): Resource requested is unreadable, please reduce your request rate"
}
(1/16) Running healthcheck tests ... done in 0 seconds
(2/16) Running mc tests ... FAILED in 4 seconds
{
  "name": "mc",
  "duration": "4232",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-24756 --quiet --no-color rb myminio/mc-test-bucket-13528  >>> \nmc: <ERROR> Failed to remove `myminio/mc-test-bucket-13528`. Resource requested is unreadable, please reduce your request rate."
}
(2/16) Running minio-dotnet tests ... done in 34 seconds
(3/16) Running minio-go tests ... FAILED in 18 seconds
{
  "args": {
    "bucketName": "minio-go-test-jrofn1jcg52nn9v2",
    "region": "eu-west-1"
  },
  "duration": 3909,
  "error": "Resource requested is unreadable, please reduce your request rate",
  "function": "MakeBucket(bucketName, region)",
  "message": "CleanupBucket failed while removing bucket recursively",
  "name": "minio-go: testMakeBucketRegionsV2",
  "status": "FAIL"
}
(3/16) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 453,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = SlowDown, message = Resource requested is unreadable, please reduce your request rate, bucketName = minio-java-test-32je3l5, objectName = null, resource = /minio-java-test-32je3l5, requestId = 1690258040BB6B04, hostId = f8249e62-3420-4382-b112-4f8a1dd0dbb2)\nrequest={method=DELETE, url=http://minio-k8s2:32551/minio-java-test-32je3l5, headers=Host: minio-k8s2:32551\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.2.2\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210709T142458Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210709/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 344\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 1690258040BB6B04\nX-Xss-Protection: 1; mode=block\nDate: Fri, 09 Jul 2021 14:24:58 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:667), io.minio.S3Base.execute(S3Base.java:460), io.minio.S3Base.executeDelete(S3Base.java:767), io.minio.MinioClient.removeBucket(MinioClient.java:1600), FunctionalTest.testBucketApi(FunctionalTest.java:459), FunctionalTest.testBucketApiCases(FunctionalTest.java:470), FunctionalTest.makeBucket(FunctionalTest.java:501), FunctionalTest.runBucketTests(FunctionalTest.java:3645), FunctionalTest.runTests(FunctionalTest.java:3719), FunctionalTest.main(FunctionalTest.java:3864)]"
}
(3/16) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 13,
  "status": "FAIL",
  "error": "S3Error: Resource requested is unreadable, please reduce your request rate at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:79:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:665:10) at endReadableNT (internal/streams/readable.js:1317:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(3/16) Running minio-py tests ... FAILED in 10 minutes and 0 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-ab46e1e4-f277-4988-b646-2552244bb8ad",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='minio-k8s2', port=32551): Max retries exceeded with url: /minio-py-test-ab46e1e4-f277-4988-b646-2552244bb8ad (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 173, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.remove_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 690, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 422, in _execute\n    no_body_trace=no_body_trace,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 285, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 574, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s2', port=32551): Max retries exceeded with url: /minio-py-test-ab46e1e4-f277-4988-b646-2552244bb8ad (Caused by ResponseError('too many 503 error responses',))\n",
  "function": "remove_bucket(bucket_name)",
  "duration": 600538
}
(3/16) Running s3cmd tests ... FAILED in 46 seconds
{
  "name": "s3cmd",
  "duration": "45610",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (SlowDown): Resource requested is unreadable, please reduce your request rate"
}
(3/16) Running s3select tests ... FAILED in 10 minutes and 2 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-48bca1ee-e36a-4443-81f7-d1765405817a"
  },
  "duration": 601735,
  "message": "HTTPConnectionPool(host='minio-k8s2', port=32551): Max retries exceeded with url: /s3select-test-48bca1ee-e36a-4443-81f7-d1765405817a (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 120, in test_csv_input_custom_quote_char\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 690, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 422, in _execute\n    no_body_trace=no_body_trace,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 285, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 574, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s2', port=32551): Max retries exceeded with url: /s3select-test-48bca1ee-e36a-4443-81f7-d1765405817a (Caused by ResponseError('too many 503 error responses',))\n",
  "status": "FAIL"
}
(3/16) Running security tests ... done in 0 seconds
(4/16) Running versioning tests ... FAILED in 32 minutes and 57 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-gp45xqf3uogzqe"
  },
  "duration": 1976348,
  "function": "/mint/build/versioning/main.go:79: main.cleanupBucket",
  "message": "Unable to cleanup bucket after compliance tests",
  "name": "versioning",
  "status": "FAIL"
}

Executed 3 out of 16 tests successfully.
```
<a name="4b2392c7-44ab-485c-9205-a0e2eceba68d"></a>
### 12659-0b4ef68/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s2:31611
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/16) Running aws-sdk-go tests ... FAILED in 6 minutes and 5 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-mkx0nd1vprrxsn",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364760,
  "error": "SlowDown: Resource requested is unreadable, please reduce your request rate\n\tstatus code: 503, request id: 16902567AD628941, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go DeleteBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/16) Running aws-sdk-java tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-mkx0nd1vprrxsn",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364760,
  "error": "SlowDown: Resource requested is unreadable, please reduce your request rate\n\tstatus code: 503, request id: 16902567AD628941, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go DeleteBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/16) Running aws-sdk-php tests ... FAILED in 5 seconds
{
  "name": "aws-sdk-php",
  "function": "getBucketPolicy ( array $params = [] )",
  "args": {
    "Bucket": "aws-sdk-php-60e85bd3f0851"
  },
  "duration": "218",
  "status": "FAIL",
  "error": "Error executing \"DeleteBucket\" on \"http://minio-k8s2:31611/aws-sdk-php-60e85bd3f0851\"; AWS HTTP error: Server error: `DELETE http://minio-k8s2:31611/aws-sdk-php-60e85bd3f0851` resulted in a `503 Service Unavailable` response:\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>SlowDown</Code><Message>Resource requested is unreadable, please red (truncated...)\n SlowDown (server): Resource requested is unreadable, please reduce your request rate - <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>SlowDown</Code><Message>Resource requested is unreadable, please reduce your request rate</Message><BucketName>aws-sdk-php-60e85bd3f0851</BucketName><Resource>/aws-sdk-php-60e85bd3f0851</Resource><RequestId>169025697171C17F</RequestId><HostId>fc70b08c-a17e-4bd1-b8aa-34456efea67c</HostId></Error>"
}
(1/16) Running aws-sdk-ruby tests ... FAILED in 3 seconds
{
  "name": "aws-sdk-ruby",
  "function": "removeBucket(bucket_name)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-d45f5d62731e"
  },
  "duration": 2306.32,
  "error": "Failed to clean-up bucket 'aws-sdk-ruby-bucket-d45f5d62731e', Resource requested is unreadable, please reduce your request rate",
  "status": "FAIL"
}
(1/16) Running awscli tests ... FAILED in 16 seconds
{
  "name": "awscli",
  "duration": 15011,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "remove_bucket failed: s3://awscli-mint-test-bucket-18570 An error occurred (SlowDown) when calling the DeleteBucket operation (reached max retries: 4): Resource requested is unreadable, please reduce your request rate"
}
(1/16) Running healthcheck tests ... done in 1 seconds
(2/16) Running mc tests ... FAILED in 5 seconds
{
  "name": "mc",
  "duration": "5091",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-11970 --quiet --no-color rb myminio/mc-test-bucket-14145  >>> \nmc: <ERROR> Failed to remove `myminio/mc-test-bucket-14145`. Resource requested is unreadable, please reduce your request rate."
}
(2/16) Running minio-dotnet tests ... done in 35 seconds
(3/16) Running minio-go tests ... FAILED in 20 seconds
{
  "args": {
    "bucketName": "minio-go-test-qqeodims1eccf3wz",
    "region": "eu-west-1"
  },
  "duration": 4299,
  "error": "Resource requested is unreadable, please reduce your request rate",
  "function": "MakeBucket(bucketName, region)",
  "message": "CleanupBucket failed while removing bucket recursively",
  "name": "minio-go: testMakeBucketRegionsV2",
  "status": "FAIL"
}
(3/16) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 510,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = SlowDown, message = Resource requested is unreadable, please reduce your request rate, bucketName = minio-java-test-1dsb17d, objectName = null, resource = /minio-java-test-1dsb17d, requestId = 1690257C4281468B, hostId = fc70b08c-a17e-4bd1-b8aa-34456efea67c)\nrequest={method=DELETE, url=http://minio-k8s2:31611/minio-java-test-1dsb17d, headers=Host: minio-k8s2:31611\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.2.2\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210709T142441Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210709/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 344\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 1690257C4281468B\nX-Xss-Protection: 1; mode=block\nDate: Fri, 09 Jul 2021 14:24:41 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:667), io.minio.S3Base.execute(S3Base.java:460), io.minio.S3Base.executeDelete(S3Base.java:767), io.minio.MinioClient.removeBucket(MinioClient.java:1600), FunctionalTest.testBucketApi(FunctionalTest.java:459), FunctionalTest.testBucketApiCases(FunctionalTest.java:470), FunctionalTest.makeBucket(FunctionalTest.java:501), FunctionalTest.runBucketTests(FunctionalTest.java:3645), FunctionalTest.runTests(FunctionalTest.java:3719), FunctionalTest.main(FunctionalTest.java:3864)]"
}
(3/16) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 13,
  "status": "FAIL",
  "error": "S3Error: Resource requested is unreadable, please reduce your request rate at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:79:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:665:10) at endReadableNT (internal/streams/readable.js:1317:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(3/16) Running minio-py tests ... FAILED in 10 minutes and 1 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-d1ca9303-94b2-4f8f-891b-9e56396c55dd",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='minio-k8s2', port=31611): Max retries exceeded with url: /minio-py-test-d1ca9303-94b2-4f8f-891b-9e56396c55dd (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 173, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.remove_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 690, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 422, in _execute\n    no_body_trace=no_body_trace,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 285, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 574, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s2', port=31611): Max retries exceeded with url: /minio-py-test-d1ca9303-94b2-4f8f-891b-9e56396c55dd (Caused by ResponseError('too many 503 error responses',))\n",
  "function": "remove_bucket(bucket_name)",
  "duration": 600562
}
(3/16) Running s3cmd tests ... FAILED in 45 seconds
{
  "name": "s3cmd",
  "duration": "45636",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (SlowDown): Resource requested is unreadable, please reduce your request rate"
}
(3/16) Running s3select tests ... FAILED in 10 minutes and 2 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-d646df83-0149-4e8b-ba66-deafdf72d0a9"
  },
  "duration": 601698,
  "message": "HTTPConnectionPool(host='minio-k8s2', port=31611): Max retries exceeded with url: /s3select-test-d646df83-0149-4e8b-ba66-deafdf72d0a9 (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 120, in test_csv_input_custom_quote_char\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 690, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 422, in _execute\n    no_body_trace=no_body_trace,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 285, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 574, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s2', port=31611): Max retries exceeded with url: /s3select-test-d646df83-0149-4e8b-ba66-deafdf72d0a9 (Caused by ResponseError('too many 503 error responses',))\n",
  "status": "FAIL"
}
(3/16) Running security tests ... done in 0 seconds
(4/16) Running versioning tests ... FAILED in 32 minutes and 57 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-hxekvepggqo65d"
  },
  "duration": 1976303,
  "function": "/mint/build/versioning/main.go:79: main.cleanupBucket",
  "message": "Unable to cleanup bucket after compliance tests",
  "name": "versioning",
  "status": "FAIL"
}

Executed 3 out of 16 tests successfully.
```
<a name="37effcae-a46d-4a76-8b9a-ad9d34cdd963"></a>
### 12659-0b4ef68/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s7:32484
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/16) Running aws-sdk-go tests ... FAILED in 6 minutes and 5 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-ezt29r40xs3zrp",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364505,
  "error": "SlowDown: Resource requested is unreadable, please reduce your request rate\n\tstatus code: 503, request id: 16902577E91504D5, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go DeleteBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/16) Running aws-sdk-java tests ... FAILED in 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-ezt29r40xs3zrp",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364505,
  "error": "SlowDown: Resource requested is unreadable, please reduce your request rate\n\tstatus code: 503, request id: 16902577E91504D5, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go DeleteBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/16) Running aws-sdk-php tests ... FAILED in 3 seconds
{
  "name": "aws-sdk-php",
  "function": "getBucketPolicy ( array $params = [] )",
  "args": {
    "Bucket": "aws-sdk-php-60e85c1a5dd25"
  },
  "duration": "175",
  "status": "FAIL",
  "error": "Error executing \"DeleteBucket\" on \"http://minio-k8s7:32484/aws-sdk-php-60e85c1a5dd25\"; AWS HTTP error: Server error: `DELETE http://minio-k8s7:32484/aws-sdk-php-60e85c1a5dd25` resulted in a `503 Service Unavailable` response:\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>SlowDown</Code><Message>Resource requested is unreadable, please red (truncated...)\n SlowDown (server): Resource requested is unreadable, please reduce your request rate - <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>SlowDown</Code><Message>Resource requested is unreadable, please reduce your request rate</Message><BucketName>aws-sdk-php-60e85c1a5dd25</BucketName><Resource>/aws-sdk-php-60e85c1a5dd25</Resource><RequestId>16902579A6525864</RequestId><HostId>7b45e4cb-956b-42e5-a3e1-c21a0051d233</HostId></Error>"
}
(1/16) Running aws-sdk-ruby tests ... FAILED in 3 seconds
{
  "name": "aws-sdk-ruby",
  "function": "removeBucket(bucket_name)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-0d63263c81df"
  },
  "duration": 2288.13,
  "error": "Failed to clean-up bucket 'aws-sdk-ruby-bucket-0d63263c81df', Resource requested is unreadable, please reduce your request rate",
  "status": "FAIL"
}
(1/16) Running awscli tests ... FAILED in 21 seconds
{
  "name": "awscli",
  "duration": 19988,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "remove_bucket failed: s3://awscli-mint-test-bucket-4637 An error occurred (SlowDown) when calling the DeleteBucket operation (reached max retries: 4): Resource requested is unreadable, please reduce your request rate"
}
(1/16) Running healthcheck tests ... done in 0 seconds
(2/16) Running mc tests ... FAILED in 4 seconds
{
  "name": "mc",
  "duration": "3697",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-2185 --quiet --no-color rb myminio/mc-test-bucket-666  >>> \nmc: <ERROR> Failed to remove `myminio/mc-test-bucket-666`. Resource requested is unreadable, please reduce your request rate."
}
(2/16) Running minio-dotnet tests ... done in 33 seconds
(3/16) Running minio-go tests ... FAILED in 24 seconds
{
  "args": {
    "bucketName": "minio-go-test-039koc0uy6uu1xnx",
    "region": "eu-west-1"
  },
  "duration": 4134,
  "error": "Resource requested is unreadable, please reduce your request rate",
  "function": "MakeBucket(bucketName, region)",
  "message": "CleanupBucket failed while removing bucket recursively",
  "name": "minio-go: testMakeBucketRegionsV2",
  "status": "FAIL"
}
(3/16) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 460,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = SlowDown, message = Resource requested is unreadable, please reduce your request rate, bucketName = minio-java-test-opk03l, objectName = null, resource = /minio-java-test-opk03l, requestId = 1690258E21754690, hostId = 7b45e4cb-956b-42e5-a3e1-c21a0051d233)\nrequest={method=DELETE, url=http://minio-k8s7:32484/minio-java-test-opk03l, headers=Host: minio-k8s7:32484\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.2.2\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210709T142557Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210709/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 342\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 1690258E21754690\nX-Xss-Protection: 1; mode=block\nDate: Fri, 09 Jul 2021 14:25:57 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:667), io.minio.S3Base.execute(S3Base.java:460), io.minio.S3Base.executeDelete(S3Base.java:767), io.minio.MinioClient.removeBucket(MinioClient.java:1600), FunctionalTest.testBucketApi(FunctionalTest.java:459), FunctionalTest.testBucketApiCases(FunctionalTest.java:470), FunctionalTest.makeBucket(FunctionalTest.java:501), FunctionalTest.runBucketTests(FunctionalTest.java:3645), FunctionalTest.runTests(FunctionalTest.java:3719), FunctionalTest.main(FunctionalTest.java:3864)]"
}
(3/16) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 13,
  "status": "FAIL",
  "error": "S3Error: Resource requested is unreadable, please reduce your request rate at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:79:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:665:10) at endReadableNT (internal/streams/readable.js:1317:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(3/16) Running minio-py tests ... FAILED in 10 minutes and 1 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-1de0d1e9-9a88-4479-88e8-1245c83cf916",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='minio-k8s7', port=32484): Max retries exceeded with url: /minio-py-test-1de0d1e9-9a88-4479-88e8-1245c83cf916 (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 173, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.remove_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 690, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 422, in _execute\n    no_body_trace=no_body_trace,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 285, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 574, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s7', port=32484): Max retries exceeded with url: /minio-py-test-1de0d1e9-9a88-4479-88e8-1245c83cf916 (Caused by ResponseError('too many 503 error responses',))\n",
  "function": "remove_bucket(bucket_name)",
  "duration": 600634
}
(3/16) Running s3cmd tests ... FAILED in 46 seconds
{
  "name": "s3cmd",
  "duration": "45638",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (SlowDown): Resource requested is unreadable, please reduce your request rate"
}
(3/16) Running s3select tests ... FAILED in 10 minutes and 2 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-20efe2e8-ec3a-4ed0-ac17-844fbf19e6b3"
  },
  "duration": 601464,
  "message": "HTTPConnectionPool(host='minio-k8s7', port=32484): Max retries exceeded with url: /s3select-test-20efe2e8-ec3a-4ed0-ac17-844fbf19e6b3 (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 120, in test_csv_input_custom_quote_char\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 690, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 422, in _execute\n    no_body_trace=no_body_trace,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 285, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 574, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s7', port=32484): Max retries exceeded with url: /s3select-test-20efe2e8-ec3a-4ed0-ac17-844fbf19e6b3 (Caused by ResponseError('too many 503 error responses',))\n",
  "status": "FAIL"
}
(3/16) Running security tests ... done in 0 seconds
(4/16) Running versioning tests ... FAILED in 32 minutes and 57 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-3tg0ofaw4z0gmj"
  },
  "duration": 1977233,
  "function": "/mint/build/versioning/main.go:79: main.cleanupBucket",
  "message": "Unable to cleanup bucket after compliance tests",
  "name": "versioning",
  "status": "FAIL"
}

Executed 3 out of 16 tests successfully.
```
<a name="2f48a660-9f14-4767-93a8-ac244be8ff74"></a>
### 12659-0b4ef68/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s5:31843
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/16) Running aws-sdk-go tests ... FAILED in 6 minutes and 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-0mefiykju9hmxp",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 365886,
  "error": "SlowDown: Resource requested is unreadable, please reduce your request rate\n\tstatus code: 503, request id: 1690257820CA2C4E, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go DeleteBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/16) Running aws-sdk-java tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-0mefiykju9hmxp",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 365886,
  "error": "SlowDown: Resource requested is unreadable, please reduce your request rate\n\tstatus code: 503, request id: 1690257820CA2C4E, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go DeleteBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/16) Running aws-sdk-php tests ... FAILED in 4 seconds
{
  "name": "aws-sdk-php",
  "function": "getBucketPolicy ( array $params = [] )",
  "args": {
    "Bucket": "aws-sdk-php-60e85c1ae851d"
  },
  "duration": "315",
  "status": "FAIL",
  "error": "Error executing \"DeleteBucket\" on \"http://minio-k8s5:31843/aws-sdk-php-60e85c1ae851d\"; AWS HTTP error: Server error: `DELETE http://minio-k8s5:31843/aws-sdk-php-60e85c1ae851d` resulted in a `503 Service Unavailable` response:\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>SlowDown</Code><Message>Resource requested is unreadable, please red (truncated...)\n SlowDown (server): Resource requested is unreadable, please reduce your request rate - <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>SlowDown</Code><Message>Resource requested is unreadable, please reduce your request rate</Message><BucketName>aws-sdk-php-60e85c1ae851d</BucketName><Resource>/aws-sdk-php-60e85c1ae851d</Resource><RequestId>16902579AE41DE87</RequestId><HostId>43d149cf-4729-4046-a54a-4de6a20f96d6</HostId></Error>"
}
(1/16) Running aws-sdk-ruby tests ... FAILED in 2 seconds
{
  "name": "aws-sdk-ruby",
  "function": "removeBucket(bucket_name)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-1d94b6ead4a4"
  },
  "duration": 2261.6,
  "error": "Failed to clean-up bucket 'aws-sdk-ruby-bucket-1d94b6ead4a4', Resource requested is unreadable, please reduce your request rate",
  "status": "FAIL"
}
(1/16) Running awscli tests ... FAILED in 20 seconds
{
  "name": "awscli",
  "duration": 19325,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "remove_bucket failed: s3://awscli-mint-test-bucket-27094 An error occurred (SlowDown) when calling the DeleteBucket operation (reached max retries: 4): Resource requested is unreadable, please reduce your request rate"
}
(1/16) Running healthcheck tests ... done in 1 seconds
(2/16) Running mc tests ... FAILED in 5 seconds
{
  "name": "mc",
  "duration": "5230",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-14751 --quiet --no-color rb myminio/mc-test-bucket-26635  >>> \nmc: <ERROR> Failed to remove `myminio/mc-test-bucket-26635`. Resource requested is unreadable, please reduce your request rate."
}
(2/16) Running minio-dotnet tests ... done in 33 seconds
(3/16) Running minio-go tests ... FAILED in 18 seconds
{
  "args": {
    "bucketName": "minio-go-test-2rm9pd9f3awgruqq",
    "region": "eu-west-1"
  },
  "duration": 5052,
  "error": "Resource requested is unreadable, please reduce your request rate",
  "function": "MakeBucket(bucketName, region)",
  "message": "CleanupBucket failed while removing bucket recursively",
  "name": "minio-go: testMakeBucketRegionsV2",
  "status": "FAIL"
}
(3/16) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 494,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = SlowDown, message = Resource requested is unreadable, please reduce your request rate, bucketName = minio-java-test-3sb167v, objectName = null, resource = /minio-java-test-3sb167v, requestId = 1690258CA92219C4, hostId = 43d149cf-4729-4046-a54a-4de6a20f96d6)\nrequest={method=DELETE, url=http://minio-k8s5:31843/minio-java-test-3sb167v, headers=Host: minio-k8s5:31843\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.2.2\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210709T142551Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210709/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 344\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 1690258CA92219C4\nX-Xss-Protection: 1; mode=block\nDate: Fri, 09 Jul 2021 14:25:51 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:667), io.minio.S3Base.execute(S3Base.java:460), io.minio.S3Base.executeDelete(S3Base.java:767), io.minio.MinioClient.removeBucket(MinioClient.java:1600), FunctionalTest.testBucketApi(FunctionalTest.java:459), FunctionalTest.testBucketApiCases(FunctionalTest.java:470), FunctionalTest.makeBucket(FunctionalTest.java:501), FunctionalTest.runBucketTests(FunctionalTest.java:3645), FunctionalTest.runTests(FunctionalTest.java:3719), FunctionalTest.main(FunctionalTest.java:3864)]"
}
(3/16) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 12,
  "status": "FAIL",
  "error": "S3Error: Resource requested is unreadable, please reduce your request rate at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:79:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:665:10) at endReadableNT (internal/streams/readable.js:1317:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(3/16) Running minio-py tests ... FAILED in 10 minutes and 1 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-3138199e-336f-43c3-a815-7dd829c8a824",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='minio-k8s5', port=31843): Max retries exceeded with url: /minio-py-test-3138199e-336f-43c3-a815-7dd829c8a824 (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 173, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.remove_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 690, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 422, in _execute\n    no_body_trace=no_body_trace,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 285, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 574, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s5', port=31843): Max retries exceeded with url: /minio-py-test-3138199e-336f-43c3-a815-7dd829c8a824 (Caused by ResponseError('too many 503 error responses',))\n",
  "function": "remove_bucket(bucket_name)",
  "duration": 600578
}
(3/16) Running s3cmd tests ... FAILED in 46 seconds
{
  "name": "s3cmd",
  "duration": "45611",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (SlowDown): Resource requested is unreadable, please reduce your request rate"
}
(3/16) Running s3select tests ... FAILED in 10 minutes and 1 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-8049be43-a8a9-4ec8-ac16-ff38d39a8c8e"
  },
  "duration": 601444,
  "message": "HTTPConnectionPool(host='minio-k8s5', port=31843): Max retries exceeded with url: /s3select-test-8049be43-a8a9-4ec8-ac16-ff38d39a8c8e (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 120, in test_csv_input_custom_quote_char\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 690, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 422, in _execute\n    no_body_trace=no_body_trace,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 285, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 574, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s5', port=31843): Max retries exceeded with url: /s3select-test-8049be43-a8a9-4ec8-ac16-ff38d39a8c8e (Caused by ResponseError('too many 503 error responses',))\n",
  "status": "FAIL"
}
(3/16) Running security tests ... done in 0 seconds
(4/16) Running versioning tests ... FAILED in 32 minutes and 57 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-19twjwlr5rzkk2"
  },
  "duration": 1976227,
  "function": "/mint/build/versioning/main.go:79: main.cleanupBucket",
  "message": "Unable to cleanup bucket after compliance tests",
  "name": "versioning",
  "status": "FAIL"
}

Executed 3 out of 16 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/storage-rest-server.go
+++ b/cmd/storage-rest-server.go
@@ -282,7 +282,7 @@ func (s *storageRESTServer) DeleteVolHandler(w http.ResponseWriter, r *http.Requ
 	}
 	vars := mux.Vars(r)
 	volume := vars[storageRESTVolume]
-	forceDelete := vars[storageRESTForceDelete] == "true"
+	forceDelete := r.URL.Query().Get(storageRESTForceDelete) == "true"
 	err := s.storage.DeleteVol(r.Context(), volume, forceDelete)
 	if err != nil {
 		s.writeErrorResponse(w, err)

